examples/flutter_gallery/test/live_smoketest.dart

```diff
@@ -11,30 +11,53 @@ import 'package:flutter/scheduler.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 import 'package:flutter_gallery/gallery/app.dart';
+import 'package:flutter_gallery/gallery/item.dart';
 
-/// Reports success or failure to the native code.
+// Reports success or failure to the native code.
 const MethodChannel _kTestChannel = const MethodChannel('io.flutter.demo.gallery/TestLifecycleListener');
 
+// The titles for all of the Gallery demos.
+final List<String> _kAllDemos = kAllGalleryItems.map((GalleryItem item) => item.title).toList();
+
+// We don't want to wait for animations to complete before tapping the
+// back button in the demos with these titles.
+const List<String> _kUnsynchronizedDemos = const <String>[
+  'Progress indicators',
+  'Activity Indicator',
+  'Video',
+];
+
+// These demos can't be backed out of by tapping a button whose
+// tooltip is 'Back'.
+const List<String> _kSkippedDemos = const <String>[
+  'Backdrop',
+  'Pull to refresh',
+];
+
 Future<Null> main() async {
   try {
-    runApp(const GalleryApp());
+    // Verify that _kUnsynchronizedDemos and _kSkippedDemos identify
+    // demos that actually exist.
+    assert(new Set<String>.from(_kAllDemos).containsAll(_kUnsynchronizedDemos));
+    assert(new Set<String>.from(_kAllDemos).containsAll(_kSkippedDemos));
```

Comment:
There's no guarantee that this test will always with assertions. Because this is on-device test, it can run in profile or production mode. Can you make these just normal `if` checks with explicit call to `fail`?